### DISCUSSION TITLE
When defining a unit programmatically, how does Pint decide what to use for formatting?

### DISCUSSION THREAD

**Discussion by nealkruis** [hgrecco/pint#2100] [Q&A]
I have a few common aliases and units that I use frequently:

```python
# Add new aliases
unit_registry.define("@alias inch_H2O_39F = in_H2O")
unit_registry.define("@alias ton_of_refrigeration = ton_ref")
unit_registry.define("@alias pound = lb_m")
unit_registry.define("@alias force_pound = lb_f")

# Add new units
unit_registry.define("cubic_feet_per_minute = cu_ft / min = cfm")
unit_registry.define("gallons_per_minute = gallon / min = gpm")
unit_registry.define("thermal_resistance_SI = m**2*K/W = R_value_SI = R_SI")
unit_registry.define("thermal_resistance_IP = ft**2*degR*h/Btu = R_value_IP = R_IP")
unit_registry.define("thermal_conductance_SI = W/(m**2*K) = U_factor_SI = U_SI")
unit_registry.define("thermal_conductance_IP = Btu/(ft**2*degR*h) = U_factor_IP = U_IP")
```

It's not clear to me how Pint decides what to use when formatting these units with `~P`, or how/if I can change the default behavior. For example, how can I specify:

- `thermal_resistance_SI` prints as `m²·K/W` (and not `R_value_SI`)
- `ton_ref` prints as `ton_ref` (and not `refrigeration_ton`)

**Comment by Doribelove** [comment]
`~P` combines two formatting choices: `~` uses the unit's **symbol** (falling back to its canonical name when there is no symbol), and `P` uses pretty-printing. It does not select whichever alias was used to access the unit. In a definition, the first name after the conversion is the symbol; later names are aliases. So this line makes `R_value_SI` the symbol:

```python
ureg.define("thermal_resistance_SI = m**2*K/W = R_value_SI = R_SI")
```

If you control the definition and want the exact displayed text, set the symbol when defining a **fresh** registry. For `ton_ref`, make it a separate unit name equivalent to the built-in refrigeration ton instead of only adding an alias to `refrigeration_ton`:

```python
import pint

ureg = pint.UnitRegistry(on_redefinition="raise")
ureg.define("thermal_resistance_SI = m**2*K/W = m²·K/W = R_value_SI = R_SI")
ureg.define("ton_ref = refrigeration_ton")

print(f"{1 * ureg.thermal_resistance_SI:~P}")  # 1 m²·K/W
print(f"{1 * ureg.ton_ref:~P}")                # 1 ton_ref
```

I tested this against current Pint source. Do not also run `@alias ton_of_refrigeration = ton_ref` in that registry: it would bind `ton_ref` as an alias first, colliding with the new unit name. If the definitions are already loaded and you only need a particular label in one report, the simplest alternative is to convert to the intended unit and format its magnitude with an explicit display label. That keeps display policy separate from Pint's unit names.

See Pint's [format specification](https://pint.readthedocs.io/en/stable/user/formatting.html) and [unit definition guidance](https://pint.readthedocs.io/en/stable/advanced/defining.html).
